lib/nano.js

```diff
@@ -705,6 +842,14 @@ module.exports = exports = function dbScope (cfg) {
         callback = body
         body = undefined
       }
+      if (!ddoc || !viewName || !docName) {
```

Comment:
Since you do the same check in `showDoc` you could probably move this check into another function. Or even do something like:

```js

checkParams = (params, callback, e) => {
     if (params.some(param => !param) {
           if (callback) {
              return callback(e, null)
        } else {
              return Promise.reject(e)
        }
     }
)}
```

That is a little rough but could work. And should simplify all your checks a bit. But you also welcome to leave it as is.

Reply:
implemented some helper functions as suggested - 👍